Client/Ecosia/AppDelegate/Leaves/TabManagerAppDelegate.swift

```diff
@@ -0,0 +1,32 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Storage
+
+final class TabManagerAppDelegate: AppDelegateLeaf {
+    
+    private var tabManager: TabManager?
+    private var profile: Profile
+    
+    required init(profile: Profile, tabManager: TabManager?) {
```

Comment:
Don't you actually need to assign `self.tabManager` in this init since it is actually initialised by the factory? 

It also feels a bit strange that this delegate is responsible for the "lifecycle" of the `tabManager` but not its initialisation... I see it is transparent with the extension being inside this file but just find it strange overall... Thought about moving this to the root composite class but can also see I'm suggesting that too much and it might be back to a huge AppDelegate class 🙈  In the end, I'm not sure I have a solution for it so would be ok about leaving it for now 😆

Reply:
The `tabManager` itself is part of those sort of compromises I had to make. Annoyingly, the side effect is some oddly references and initialisation.